faucet/src/faucet.rs

```diff
@@ -504,28 +504,40 @@ async fn request_fee_assets(
 
 async fn worker(id: usize, mut state: FaucetState) {
     'wait_for_requests: while let Some(pub_key) = state.queue.pop().await {
-        let mut wallet = state.wallet.lock().await;
-        let faucet_addr = wallet.pub_keys().await[0].address();
-
         loop {
-            tracing::info!(
-                "worker {}: transferring {} tokens from {} to {}",
-                id,
-                state.grant_size,
-                net::UserAddress(faucet_addr.clone()),
-                net::UserAddress(pub_key.address())
-            );
-            let balance = wallet.balance(&AssetCode::native()).await;
-            let records = spendable_records(&wallet, state.grant_size).await.count();
-            tracing::info!(
-                "worker {}: wallet balance before transfer: {} across {} records",
-                id,
-                balance,
-                records
-            );
+            // If we don't have a sufficient balance, to transfer, it is probably only because some
```

Comment:
Is there not a risk to have this loop go on for ever? For example if the transactions in flight are rejected or if someone uses the same wallet somewhere else.

Reply:
I think it's okay. This worker task is supposed to live forever anyways. And as long as the balance is too low, it would not be able to make progress no matter what. If we allowed it to continue, it would only fail to transfer due to insufficient balance and send the failed request to the back of the queue.

Also, if the transactions in flight are rejected, we would get their inputs back, which would also allow this loop to exit.